docs/BUILD.md

```diff
@@ -112,13 +112,17 @@ s2n-tls has a dependency on a libcrypto library. A supported libcrypto must be l
 - [AWS-LC](https://github.com/aws/aws-lc)
   - Limited ["Sandboxing"](https://github.com/aws/aws-lc/blob/main/SANDBOXING.md) is only supported and tested with AWS-LC.
   - [PQ key exchange](https://aws.github.io/s2n-tls/usage-guide/ch15-post-quantum.html) is only supported with AWS-LC.
+  - FIPS mode is supported when s2n-tls is linked with a FIPS validated version of [AWS-LC](https://github.com/aws/aws-lc/blob/main/crypto/fipsmodule/FIPS.md).
 - [OpenSSL](https://www.openssl.org/) (versions 1.0.2 - 3.0)
   - ChaChaPoly is not supported before Openssl-1.1.1.
   - RSA-PSS is not supported before Openssl-1.1.1.
   - RC4 is not supported with Openssl-3.0 or later.
+  - FIPS mode is not supported with Openssl.
 - [BoringSSL](https://boringssl.googlesource.com/boringssl)
   - OCSP features are not supported with BoringSSL.
+  - FIPS mode is not supported with BoringSSL.
 - [LibreSSL](https://www.libressl.org/)
+  - FIPS mode is not supported with LibreSSL.
```

Comment:
If BoringSSL and LibreSSL don't offer a FIPS version, we probably don't need to specify that they don't support FIPS. I only thought it was worth listing for OpenSSL because we COULD support FIPS mode for openssl-3.0, but don't.

Reply:
BoringSSL does offer a fipsmodule. https://boringssl.googlesource.com/boringssl/+/master/crypto/fipsmodule/FIPS.md

LibreSSL is not FIPS certified so I removed that line.